data_import/tga_importer.py

```
@@ -152,33 +137,33 @@ def _detect_manufacturer(file_path: str) -> str:
             return "Netzsch"
-        elif "Time (s)" in header and "Furnace Temperature (°C)" in header:
+        elif "Setaram" in header or ("Time (s)" in header and "Furnace Temperature (°C)" in header):
             return "Setaram"
         else:
             raise ValueError("Unable to detect manufacturer automatically. Please specify manually.")
     except FileNotFoundError:
         logger.error(f"File not found: {file_path}")
         raise
+    except Exception as e:
+        logger.error(f"Error detecting manufacturer: {str(e)}")
+        raise ValueError(f"Unable to detect manufacturer. Error: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)